include/crocoddyl/core/optctrl/shooting.hxx

```diff
@@ -154,6 +154,9 @@ Scalar ShootingProblemTpl<Scalar>::calc(const std::vector<VectorXs>& xs, const s
   terminal_model_->calc(terminal_data_, xs.back());
 
   cost_ = Scalar(0.);
+#ifdef CROCODDYL_WITH_MULTITHREADING
+#pragma omp simd reduction(+ : cost_)
```

Comment:
Could you please point me to a reference where I could see the documentation of simd reduction clause?

Reply:
- https://docs.microsoft.com/en-us/cpp/parallel/openmp/openmp-simd?view=msvc-160
- https://www.ibm.com/support/knowledgecenter/en/SSXVZZ_16.1.0/com.ibm.xlcpp161.lelinux.doc/compiler_ref/prag_omp_simd.html

Reply:
I see a little of improvement with these routines.
Perhaps, you could try it for double checking this.

Reply:
You cannot use SIMD here, this is not aligned in memory the cost.

Reply:
From what I understand after reading the documentation is, SIMD directives parallelize the execution intra-thread, by using vectorization of the associated variables. 

What I don't understand is, how this worked for a scalar where such a vectorization isn't possible. Could you please post the timings that you saw here?

Reply:
> You cannot use SIMD here, this is not aligned in memory the cost.

Cost is an scalar, why do you think we cannot use it? (I did not observe any issue despite that I spend all day running this code with a MPC code.

> What I don't understand is, how this worked for a scalar where such a vectorization isn't possible.

This is indeed not explained, but I got inspiration by this tutorial: https://bisqwit.iki.fi/story/howto/openmp/#SimdConstructOpenmp%204%200 (see The reduction clause).

> Could you please post the timings that you saw here?

I observed a time reduction of 0.02-0.09 ms for a single iteration of the FDDP solver.
I run this benchmark for the ANYmal case with 8 threads.

Reply:
> Cost is a scalar, why do you think we cannot use it? (I did not observe any runtime issue despite that I spend all day running this code with a MPC code.

The vectorisation works for 2D vectors and more.

Reply:
> The vectorisation works for 2D vectors and more.

Yes, that's correct! But in this case, we still have multiple data.
And this macro works for such cases as described in the this link:  https://bisqwit.iki.fi/story/howto/openmp/#SimdConstructOpenmp%204%200

Reply:
Since there is a minor performance improvement, it would be worthwhile to keep the simd construct here. Thanks for this addition carlos. I'll resolve this discussion, please feel free to re-open it if some issue still remains.